packages/components/select/src/types.ts

```diff
@@ -170,7 +170,7 @@ export type SelectOptionGroupComponent = FunctionalComponent<
 
 export type SelectData = SelectOptionProps | SelectOptionGroupProps
 
-export type SelectSearchFn = (data: SelectData, searchText: string) => boolean
+export type SelectSearchFn = (data: SelectData, searchText: string, type?: string) => boolean
```

Comment:
```suggestion
export type SelectSearchFn = (data: SelectData, searchText: string) => boolean
```
可以通过 data 有没有 children 来判断，不需要额外再给一个参数。